lib/core/src/Cardano/Wallet/Checkpoints.hs

```diff
@@ -132,3 +145,120 @@ instance Buildable (DeltaCheckpoints a) where
     build (PutCheckpoint slot _) = "PutCheckpoint " <> build slot
     build (RollbackTo slot) = "RollbackTo " <> build slot
     build (RestrictTo slots) = "RestrictTo " <> listF slots
+
+{-------------------------------------------------------------------------------
+    Checkpoint hygiene
+-------------------------------------------------------------------------------}
+-- | Storing EVERY checkpoints in the database is quite expensive and useless.
```

Comment:
well, I agree in 100% it is costly, but not 100% that it is useless - what about blockchain having 3 checkpoints debt ? :trollface:

Reply:
I will rephrase this comment from the past in a future pull request. 🤓